src/kernels/jupyter/serverSelector.ts

```diff
@@ -177,9 +146,9 @@ export class JupyterServerSelector {
             placeholder,
             items,
             activeItem,
-            title: allowLocal
-                ? DataScience.jupyterSelectURIQuickPickTitle()
-                : DataScience.jupyterSelectURIQuickPickTitleRemoteOnly(),
+            acceptFilterBoxTextAsSelection: true,
```

Comment:
Since URI entering happens on the quick pick, we can't use input box validation. So I added this to allow a modal dialog to appear.